### PR TITLE
Switch to Rosti & Poetry, update dependencies

### PR DESCRIPTION
**Pull request by encukou**
Heroku will stop free hosting soon (#50), so this app needs an update. This PR has a few more updates than strictly necessary – see individual commits.

My branch is now deployed with [rosti.cz](Roští) at https://zpetnovazebnik.pyladies.cz/ with a slightly modified **copy** of the data, which will be discarded later. Your password should work there, so feel free to test things out (but be sure you're not on messing with the „live” app at herokuapp.com).

Heroku deploys automatically from `master`, but this PR is for the `main` branch so merging this shouldn't break Heroku.
The plan is:

- [x] Put the Heroku app into [maintenance mode](https://devcenter.heroku.com/articles/maintenance-mode)
- [x] Copy all data to the new app
- [x] Merge this PR
- [x] Replace the Heroku app with [heroku-redirect](https://github.com/kenmickles/heroku-redirect), and take it out of maintenance mode
- [x] Update this repo to set `main` as the default branch
- [x] Update `rosti` to pull from this repo (but note that deployment there is not automatic like with Heroku).

### CODE REVIEW

File: pyproject.toml [hunk 1]
```diff
@@ -0,0 +1,25 @@
+[tool.poetry]
+name = "zpetnovazebnik"
+version = "0.1.0"
+description = ""
+authors = ["Your Name <you@example.com>"]
```
Comment (befeleme) [inline pyproject.toml:5]: Do you want to keep the placeholder?

File: README.md [hunk 2]
```diff
@@ -35,6 +35,29 @@ prostředí dají spustit pomocí:
 $ poetry run python -m pytest
 ```
 
+## Nasazení
+
+Aplikace jede na [rosti.cz](https://rosti.cz/). Pro nasazení je potřeba:
```
Comment (befeleme) [inline README.md:40]: It's great to have the instructions about the deployment, however I miss the information who to talk to about deployment. Is it only you or is there a bigger group of content curators? Should we mention it here?

**Comment by befeleme** [comment]
Thank you for all those necessary changes. I did try to run Zpetnovazebnik as it is now and experienced all those problems addressed in the commits. The current setup works like charm. What I checked:
- tested the instructions - I find them complete except the info about the deployment rights
- ran the tests - experienced a lot of `Django50DeprecationWarnings` in the current version - all addressed in this PR, tests run without issues
- ran the local server and verified the app is working
- played with the admin of the app deployed on Rosti - its UX is nicer in this Django version; adding and removal of courses and sessions works as expected, it's possible to add a new comment in the web app - no regressions spotted

The migration plan sounds good. How long will the app be down? When will you have time to perform the migration?

File: pyproject.toml (hunk 1 above)
Comment (encukou) [reply pyproject.toml]: Oh, didn't notice that. Guess it should be `PyLadies CZ`.

File: README.md (hunk 2 above)
Comment (encukou) [reply README.md:40]: It's a bigger group but I don't know thde details. I opened https://github.com/pyvec/docs.pyvec.org/issues/288

**Comment by encukou** [comment]
Thanks!

Switching the database should take a few minutes.
I don't know how long it'll take to set up the redirection -- I've never done that. I don't think it'll be more than an hour.

I could do it today, or Monday. Whatever is OK with you (as you run the only currently active course).

**Review by befeleme — APPROVED** [review]
Monday sounds fine! Please merge as you see fit. (I know it shouldn't break anything, but maybe it's safer to do it just before the migration ;))

**Comment by encukou** [comment]
Good point. I'll merge after the site is in maintenance mode. I updated the list above.
Starting now.

**Comment by encukou** [comment]
One more thing I did is set branch protection for the old `master` branch. It now requires the maximum of 6 reviews, so hopefully no one will merge to it (which might still trigger something, maybe).

**Pull request merged**